lifecycle/lifecycle/job/portfolio.py

```diff
@@ -0,0 +1,72 @@
+from racetrack_client.utils.semver import SemanticVersion
+from racetrack_client.utils.time import days_ago
+from racetrack_commons.entities.dto import JobDto, JobStatus
+
+
+def enrich_jobs_purge_info(jobs: list[JobDto]) -> list[dict]:
+    """Enrich jobs with removal candidates info: score and reasons"""
+    job_dicts = []
+    for job in jobs:
+        score, reasons = assess_job_usability(job, jobs)
+        job_dict = job.dict()
+        job_dict['purge_score'] = -score
+        job_dict['purge_reasons'] = '\n'.join(reasons)
+        job_dict['purge_newer_versions'] = _count_job_newer_versions(job, jobs)
+        job_dicts.append(job_dict)
+    return sorted(job_dicts, key=lambda f: -f['purge_score'])
+
+
+def assess_job_usability(job: JobDto, all_jobs: list[JobDto]) -> tuple[float, list[str]]:
+    """
+    Assess usability of a job.
+    Return score number representing penalty points with descriptions of reasons.
+    A lower value means a better candidate for removal.
+    Positive score means the job shouldn't be removed.
+    """
+    score: float = 0
+    reasons = []
+
+    if job.status == JobStatus.LOST.value:
+        score -= 50
+        reasons.append('Job is lost - can\'t be found in a cluster. It should be redeployed or removed.')
+    elif job.status == JobStatus.ERROR.value:
+        score -= 20
+        reasons.append('Job is failing. It should be fixed or removed.')
+
+    deployed_days_ago = days_ago(job.update_time) or 0
+    if deployed_days_ago >= 1:  # wait a day until we decide it was never called
+        never_called = job.last_call_time is None or job.last_call_time == 0
+        if never_called:
+            score -= 10
+            reasons.append('Job seems to be unused as it was never called.')
+        else:
+            last_call_days_ago = days_ago(job.last_call_time)
+            if last_call_days_ago > 1:
+                score -= 10 * _interpolate(last_call_days_ago, 0, 30)
+                reasons.append(f'Job hasn\'t been called for {last_call_days_ago:.1f} days.')
+
+    newer_versions = _count_job_newer_versions(job, all_jobs)
+    if newer_versions > 0:
+        score -= 1 * newer_versions
```

Comment:
This should be interpolated as well. You shouldn't remove a legacy job (that might be legacy for a reason) with a higher priority than a lost job if it's legacy enough.

Also, you should probably be able to escape this scoring as a job somehow. It's all but guaranteed that someone somewhere will be running a mission critical system that's 10 years out of date, and that shouldn't lead to it getting axed.